ChartIQ/finsemble-fdc3
# Prevent channel message loops

A working version of the prevent message loops in FDC3. 

**What does this accomplish?**
If the Finsemble _component A_ is listening to context on a channel (groupA) and the same _component A_ broadcasts on the channel (groupA) it should not receive the context in its own context listener. It should recieve context if a second _component B_ is broadcasting on (groupA) _component A_ should receive the context.


**Testing steps:** 

1. Add the FDC3 Preload to the Welcome Component
2. Open two Welcome Components and connect them both to linker purple channel
3. On both Welcome Components open their developer console's (ctrl+shift+i)
4. In both components copy and paste the following and press enter to execute the code: `fdc3.addContextListener(console.log)`
5. In **one** of the Welcome Components add the following code `fdc3.broadcast({type:"instrument",id:{ticker:"MSFT"}})` and then press enter to broadcast context over the channel
6. You should observe the data broadcast from the Welcome Component does not appear in the console but in the other Welcome Component a message reflecting the data should appear. IF the data appears in both consoles after running the broadcast command above this is a FAIL and needs to be revisited.
